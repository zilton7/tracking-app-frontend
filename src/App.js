import Nav from "./components/Nav";
import Measurements from "./components/Measurements";

function App() {
  return (
    <div className="App">
      <div className="heading">Tracking App</div>
      <Measurements />
      <Nav />
    </div>
  );
}

export default App;
