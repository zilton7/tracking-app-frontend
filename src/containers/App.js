import Router from "../router/Router";
import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import loadMeasurements from "../actions/measurementsAction";
import Heading from "../components/Heading";
import Login from "../components/Login";

const App = () => {
  // Fetch measurements data
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(loadMeasurements());
  }, [dispatch]);

  // Load user data from state
  let user = useSelector((state) => state.user);

  let app = <Login />;
  if (user.loggedIn) {
    app = (
      <>
        <Heading />
        <Router />
      </>
    );
  }

  return <div className="App">{app}</div>;
};

export default App;
