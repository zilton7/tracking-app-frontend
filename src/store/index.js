import thunk from 'redux-thunk';
import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../reducers';

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(rootReducer, composeEnhancer(applyMiddleware(thunk)));

export default store;
