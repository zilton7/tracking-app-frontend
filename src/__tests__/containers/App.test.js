import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../store/index';
import App from '../../containers/App';

afterEach(cleanup);

it("Takes website's snapshot", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  expect(
    asFragment(
      <Provider store={store}>
        <App />
      </Provider>,
    ),
  ).toMatchSnapshot();
});

it("Displays app's title", () => {
  render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  const element = screen.getByText(/Tracking App/i);
  expect(element).toBeInTheDocument();
});

it('Displays username field', () => {
  render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  const element = screen.getByPlaceholderText(/Enter Your Name/i);
  expect(element).toBeInTheDocument();
});

it('Displays login button', () => {
  render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  const element = screen.getByText(/Login/i);
  expect(element).toBeInTheDocument();
});
