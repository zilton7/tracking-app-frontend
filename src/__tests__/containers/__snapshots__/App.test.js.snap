// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Takes website's snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="login-page"
    >
      <div
        class="container"
      >
        <h1>
          Tracking App
        </h1>
        <input
          id="username"
          placeholder="Enter Your Name"
        />
        <button
          type="submit"
        >
          Login
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
